lib/Makefile.am

```diff
@@ -78,11 +78,9 @@ tests_testRFC1738_LDADD= \
 tests_testRFC1738_LDFLAGS = $(LIBADD_DL)
 
 
-## Special Universal .h dependency test script
-## aborts if error encountered
-testHeaders: $(top_srcdir)/include/*.h
-	$(SHELL) $(top_srcdir)/test-suite/testheaders.sh "$(CXXCOMPILE)" $^ || exit 1
+## SOURCES cannot be used here: the header files we want to test are in include/
+testHeaders: $(top_srcdir)/include/*.h $(top_srcdir)/test-suite/testHeader.cc.in
+	$(MAKE) $(^:.h=.hdrtest) && cp $(TRUE) $@
```

Comment:
> I'll push back on this request. Special case by special case

I strongly disagree that replacing a special case with a general, reusable mechanism can be accurately characterized as "special case by special case". 

I believe that a habit of creating new (or not addressing old) special cases (instead of developing general/comprehensive solutions) is a primary contributor to growing technical debt because these special case leftovers tend to multiply and impede progress: The next developer will add another special case (because they were not given a reusable mechanism _and_ can successfully use the "We are already doing this!" excuse), and then we would have to keep all the special cases working when improving testHeaders...

"Overengineering" is engraved on the other side of that coin, of course. In this particular case, adding `ADD_TO_HDR_TESTING` mechanism does not feel like overengineering to me because it will have tiny effect on the existing general purpose code (AFAICT), will not complicate existing non-special cases, will be immediately used (for this specific use case), will eliminate fairly complex/tricky code duplication (and might even result in further code improvements/simplifications when there are no special cases standing in the way!).

Reply:
> > I'll push back on this request. Special case by special case
> 
> I strongly disagree that replacing a special case with a general, reusable mechanism can be accurately characterized as "special case by special case".

It might be a reusable mechanism, but right now it is used exactly once, for a use case that is
highly unlikely to be reused, and might disappear; it feels overengineered (and I see you addressed this
in your answer already)

If there were two use cases, or there was a sensible chance that a second use case might appear,
I would agree with you

> I believe that a habit of creating new (or not addressing old) special cases (instead of developing general/comprehensive solutions) is a primary contributor to growing technical debt because these special case leftovers tend to multiply and impede progress: The next developer will add another special case (because they were not given a reusable mechanism _and_ can successfully use the "We are already doing this!" excuse), and then we would have to keep all the special cases working when improving testHeaders...
> 
> "Overengineering" is engraved on the other side of that coin, of course. In this particular case, adding `ADD_TO_HDR_TESTING` mechanism does not feel like overengineering to me because it will have tiny effect on the existing general purpose code (AFAICT), will not complicate existing non-special cases, will be immediately used (for this specific use case), will eliminate fairly complex/tricky code duplication (and might even result in further code improvements/simplifications when there are no special cases standing in the way!).

However, the implementation is be trivial so I'll just do it ¯\_(ツ)_/¯.